devops-970.nix

```diff
@@ -0,0 +1,144 @@
+
+        # if stdout is a tty, then mono 5.8 will barf over the terminfo files being too new
+        # mono 5.16 supports them, but isnt in our current nixpkgs
+        # for more info, refer to `mcs/class/corlib/System/TermInfoReader.cs` and `ReadHeader`
+        echo $PASS | signcode -spc ${toString signingKeys.spc} -v ${toString signingKeys.pvk} -a sha1 -$ commercial -n "TODO description" -i http://iohk.io -t http://timestamp.verisign.com/scripts/timstamp.dll -tr 10 $FILE | cat
+        storePath=$(nix-store --add-fixed sha256 $FILE)
+        rm -rf $DIR
+        echo $storePath >&3
+      '';
+      res = builtins.exec [ signingScript ];
+    in res;
+    signedCardano = pkgs.runCommand "signed-daedalus-bridge" {} ''
+      cp -r ${self.unpackedCardano} $out
+      chmod -R +w $out
+      cd $out
+      rm *.exe
+      cp ${self.signFile "${self.unpackedCardano}/cardano-launcher.exe"} cardano-launcher.exe
+      cp ${self.signFile "${self.unpackedCardano}/cardano-node.exe"} cardano-node.exe
+      cp ${self.signFile "${self.unpackedCardano}/cardano-x509-certificates.exe"} cardano-x509-certificates.exe
+      cp ${self.signFile "${self.unpackedCardano}/wallet-extractor.exe"} wallet-extractor.exe
+    '';
+    nsisFiles = pkgs.runCommand "nsis-files" { buildInputs = [ old.daedalus-installer ]; } ''
+      mkdir installers
+      cp -vir ${./package.json} package.json
+      cp -vir ${./installers/dhall} installers/dhall
+      cd installers
+
+      make-installer --os win64 -o $out --cluster ${cluster} buildkite-cross
+
+      mkdir $out
+      cp daedalus.nsi uninstaller.nsi launcher-config.yaml wallet-topology.yaml $out/
+    '';
+    uninstaller = pkgs.runCommand "uninstaller" { buildInputs = [ self.nsis pkgs.winePackages.minimal ]; } ''
+      mkdir home
+      export HOME=$(realpath home)
+
+      ln -sv ${./installers/nsis_plugins} nsis_plugins
+      cp ${self.nsisFiles}/uninstaller.nsi .
+
+      makensis uninstaller.nsi -V4
+
+      wine tempinstaller.exe /S
+      mkdir $out
+      mv -v $HOME/.wine/drive_c/uninstall.exe $out/uninstall.exe
+    '';
+    installer = pkgs.runCommand "win64-installer-${cluster}" { buildInputs = [ old.daedalus-installer self.nsis pkgs.unzip pkgs.winePackages.minimal self.configMutator pkgs.jq old.yaml2json ]; } ''
+      mkdir home
+      export HOME=$(realpath home)
+
+      mkdir -p $out/{nix-support,cfg-files}
+      mkdir installers
+      cp ${./cardano-sl-src.json} cardano-sl-src.json
+      cp -vir ${./installers/dhall} installers/dhall
+      cp -vir ${./installers/icons} installers/icons
+      cp -vir ${./package.json} package.json
+      chmod -R +w installers
+      cd installers
+      mkdir -pv ../release/win32-x64/
+      cp -r ${old.rawapp-win64} ../release/win32-x64/Daedalus-win32-x64
+      chmod -R +w ../release/win32-x64/Daedalus-win32-x64
+      cp -v ${old.fastlist}/bin/fastlist.exe ../release/win32-x64/Daedalus-win32-x64/resources/app/dist/main/fastlist.exe
```

Comment:
@nikolaglumac, our use of https://www.npmjs.com/package/ps-list is adding significant complexity and fragility.

Do you think we can get rid of it?

Reply:
@deepfire `ps-list` was the best option at the time. I am pretty sure it is the same nowadays too. The complexity arises from the fact that it is hard to find a 100% platform compatible module. This is why @cleverca22 had to add all those `fastlist.exe` tweaks...